### ISSUE TITLE
CreditCardView not showing on NS-VUE ?

### ISSUE THREAD

**Issue by Jonarod** [triniwiz/nativescript-stripe#68]
Can't make `CreditCardView` to work with NS-VUE...
Any idea ?

```
<CreditCardView id="card" ref="card"></CreditCardView>
```

```
import { CreditCardView } from 'nativescript-stripe';
export default {
  components:{CreditCardView},
}
```
Nothing shows up on the UI...

**Comment by triniwiz** [comment]
U need to register the component

**Comment by Jonarod** [comment]
I have been trying:

```js
// main.js
import Vue from 'nativescript-vue'
Vue.registerElement('CreditCardView', () => require('nativescript-stripe').CreditCardView );
```

```html
// app.vue
<Template>
    <Page>
        <CreditCardView id="card" ref="card"></CreditCardView>
    </Page>
<Template>

<Script>
import { CreditCardView } from 'nativescript-stripe';
export default {
   components:{CreditCardView}
}
</Script>
```



to no avail... Am I not supposed to see the `CardInputWidget` ?

**EDIT**: closed the issue unintentionally...

**Comment by Jonarod** [comment]
Here is a repo to reproduce the issue: [https://github.com/Jonarod/nativescript-vue-stripe-demo](https://github.com/Jonarod/nativescript-vue-stripe-demo)

**Comment by Jonarod** [comment]
OK got it working: there is no need to register the component in the script...

```js
// main.js
import Vue from 'nativescript-vue'
Vue.registerElement('CreditCardView', () => require('nativescript-stripe').CreditCardView );
```
is enough.
Then just:

```html
<Template>
    <Page>
        <CreditCardView id="card" ref="card"></CreditCardView>
    </Page>
<Template>

<Script>
// import { CreditCardView } from 'nativescript-stripe';  NOT NECESSARY
export default {
//   components:{CreditCardView} NOT NECESSARY
}
</Script>
```

**Comment by AngeloAnolin** [comment]
Hi @Jonarod. I just revisited this, and I am wondering if you were able to perform a full implementation for NS-Vue? I have opened an issue #86 which I am unable to go further as necessary.

I looked at your [repo](https://github.com/Jonarod/nativescript-vue-stripe-demo) but I can only see the Credit Card Input field, but not the implementation on how you instantiated Stripe and execute the process to obtain Tokens. 

Thanks.

**Comment by Jonarod** [comment]
@AngeloAnolin I don't know what you mean by "full implementation", most of the Stripe workflow is made server side, not in NS-Vue...
As far as I remember, I just used this library to get the native Stripe form and validate the credit card input. From there, I just send the details to the server where it performs all the steps to process the payment.

**Comment by AngeloAnolin** [comment]
What I meant is in your sample repo, you are able to display the Credit Card Input via the markup:

```
<CreditCardView id="ccView" ref="ccView></CreditCardView>
```

But I have very little idea on where you are able to generate the tokens (i.e. sample code below):

```
generateTokens() {
  let ccView = this.$refs.ccView
  let card = ccView.card    // This results to undefined when I do a console.log
  let stripe = new Stripe('PUBLISHABLE_KEY')
  stripe.createToken(card).then((token) => {
    // Use this token and pass it to server side for further processing.
  }, (error) => {
    // Handle error here.
  }
}
```

But since my `card` variable does not relate to anything, the `stripe.createToken(card)` fails.

I also tried explicitly declaring a card like:

```
let card = new Card('4242424242424242', 11, 2022, '295')
let stripe = new Stripe('PUBLISHABLE_KEY')
stripe.createToken(card).then((token) => {   
  // Use this token and pass it to server side for further processing.
}, (error) => {
  // Handle error here.
}
```

but then the doing a `console.log('stripe' stripe)` only yields: 

`'stripe', {}`

So I am kind of stuck at the moment.

If you are using the CreditCardView markup, how were you able to peruse that to obtain the `Card` object which will allow for you to call the method `stripe.createToken(card)` method?

Appreciate your insight. Thanks.

Additional Info:
I believe the server side stuff you refer to are in this [article](https://stripe.com/docs/saving-cards).

What I need to obtain pretty much is the tokenized card info, which I am assuming would be handled by the NativeScript stripe library, as we want to make sure that we are not passing the Credit Card Information to the backend (for PCI Compliance) and instead sending a tokenized information that relates to the credit card.

**Comment by AngeloAnolin** [comment]
Just revisiting and hoping that @Jonarod can provide some insight.

You mentioned that he is basically passing the details to you server. So would you be sending the credit card info then via some service call? Are you sending the tokenized cc info or are you sending the actual cc info? Also, how are you referencing the CreditCardView to obtain the card info that was entered?

Thanks. Apologies for asking, but I was hoping you could share some info which I may find useful.

**Comment by Jonarod** [comment]
@AngeloAnolin It is hard to help you. What is your issue specifically?
Have you read this: [https://stripe.com/docs/charges](https://stripe.com/docs/charges) ?!
From what I see you already generated a token client-side based on the credit card info, so I guess you should just send the token to your backend and follow Stripe's instructions to charge customers from some backend.
Just as a reminder, the process is something like:
1. You present a form to you customer to fill in credit card info
2. You transform these info into a unique token (to prevent someone from eyedroping and catch credit card info in clear
3. Send the token to a backend YOU HAVE TO WRITE YOURSELF like described in the link above
4. From the server, charge the card

Of course there are other possibilities and routes you can choose server-side like payment intents, recurring payments etc... But these are completely out of the scope of this library.

From the code you shared, I would say that this very library already did the job, now you should just proceed with the next steps :)

**Comment by Jonarod** [comment]
@AngeloAnolin I reopened the code I used, here is what I have client-side (in the app):

```
<template>
      <StackLayout>
              <CreditCardView ref="ccView" id="ccView"/>
              <Button text="Pay" @tap="submit" />
      </StackLayout>
</template>

<script>
import { Stripe, Card } from 'nativescript-stripe';
const stripe = new Stripe('pk_test_TYooMQauvdEDq54NiTphI7jx'); // Public tests

export default {

  methods: {
    submit(){

     // TRY TO CREATE A TOKEN MANUALLY FIRST AND SEE WHAT HAPPENS
      const cardElement = new Card("4242424242424242",2,22,"123");

     // IF A TOKEN CAN BE CREATED MANUALLY, TRY TO EXTRACT CARD INFOS FROM THE CARDVIEW
      const cardElement = this.$refs.ccView.card;

      cardElement.name = "John Snow";

      stripe.createToken(cardElement, (error, token) => {
          if(error){
            console.log(error.message)
          } else {
            // Send token to backend
          }
      })
    }
}
</script>
```

Also, in the main.js, I have:
```
Vue.registerElement('CreditCardView', () => require('nativescript-stripe').CreditCardView);
```

I took it from an app in production, I just replace the token and the credit card numbers, so if it does not work for you... I don't know what to do.

**Comment by AngeloAnolin** [comment]
Hi @Jonarod,

Thanks and appreciate you for taking the time to provide a code sample.

The issue I am having at the moment is that even after instantiating the Stripe instance via the code:

```
import { Stripe, Card } from 'nativescript-stripe';
const stripe = new Stripe('pk_test_TYooMQauvdEDq54NiTphI7jx'); // Public tests
console.log('stripe', stripe) // undefined
```

Only yields undefined.

Also, even though I am getting a reference to the CreditCardView by:

```
const ccView = this.$refs.ccView
const ccCard = this.$refs.ccView.card
console.log('ccView', ccView)
console.log('ccCard', ccCard)
```

console.log(ccView) displays:
```
{ nodeType: 1,
_tagName: 'nativecreditcardview',
parentNode:
{ nodeType: 1,
_tagName: 'nativestacklayout',
parentNode:
{ nodeType: 1,
_tagName: 'nativestacklayout',
parentNode: [Object],
childNodes: [Object],
prevSibling: null,
nextSibling: null,
_ownerDocument: null,
_nativeView: [Object],
_meta: [Object],
removeAttribute: [Object],
hasAttribute: [Object] },
childNodes: [ [Object], [Object], [Circular], [Object], [length]: 4 ],
prevSibling:
{ nodeType: 1,
_tagName: 'nativestacklayout',
parentNode: [Object],
childNodes: [Object],
prevSibling: [Object],
nextSibling: [Circular],
_ownerDocument: null,
_nativeView: [Object],
_meta: [Object],
removeAttribute: [Object],
hasAttribute: [Object],
__vOriginalVisibility: 'vis<…>
```

While console.log(ccCard) shows:
`undefined`

So I am thinking I might be missing an import or implementation on this end.

Still looking for a simple resolution on this, and I am out of ideas at the moment where to look at. 

Additional Info:
I tried the following code:

```
let card = new Card('4242424242424242', 11, 22, '295')
console.log('card', card)
// console.log shows: { native: {} }
```

So I am thinking that the import {Stripe, Card} is not being imported properly.

```
import { Stripe, Card } from 'nativescript-stripe'
const stripeInstance = new Stripe('pk_test_AAAAAAA')
```

Hence, the reason why I am not getting anything out of Stripe, nor the Card instantiation.

P.S. I also even tried to perform a clean of the platform:
`rm -rf platforms/*`

to no avail..

**Comment by Jonarod** [comment]
Did you register the component in your main.js??

```
Vue.registerElement('CreditCardView', () => require('nativescript-stripe').CreditCardView);
```

**Comment by Jonarod** [comment]
Also, try to reinstall everything:

```
rm -rf node_modules
rm -rf package-lock.json
rm -rf platform
npm i
```

**Comment by AngeloAnolin** [comment]
@Jonarod 
Hold on for a bit. Seems something has changed in `tns-core-modules` after updating the CLI to 6.2.0 (https://github.com/NativeScript/nativescript-cli/releases/tag/v6.2.0) where the reference by stripe library to utilis.ios.js went missing. When I try to run the application, I am getting the following:

```
ERROR in ../node_modules/nativescript-stripe/stripe.js
Module not found: Error: Can't resolve 'tns-core-modules/utils/utils.ios' in '/Users/angelo/Projects/stripesimple/node_modules/nativescript-stripe'
 @ ../node_modules/nativescript-stripe/stripe.js 11:18-61
 @ ./main.js
```

I have logged this issue [here](https://github.com/triniwiz/nativescript-stripe/issues/87).

As for registering the CreditCardView as you have mentioned via this code to be added to main.js:

`Vue.registerElement('CreditCardView', () => require('nativescript-stripe').CreditCardView);`

I have this line of code added but still, I am not getting that particular UI element. It does display on the application though, but I am unable to reference it.

I tried different variations, such as:

```
const ccView = this.$refs.ccView.card; // undefined
const ccView2 = this.$refs.ccView.nativeView.card; // undefined
const ccView3 = this.$refs.ccView;
const card3 = ccView3.card // undefined
```

At the moment, I am trying to restore my NS CLI to an older version to make things work.
